src/rustdoc-json-types/lib.rs

```diff
@@ -281,19 +281,28 @@ pub enum StructType {
     Unit,
 }
 
+#[non_exhaustive]
+#[serde(rename_all = "snake_case")]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum Modifiers {
+    Const,
+    Unsafe,
+    Async,
+}
+
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct Function {
     pub decl: FnDecl,
     pub generics: Generics,
-    pub header: String,
+    pub header: Vec<Modifiers>,
```

Comment:
Should this be a HashSet so it's guaranteed not to have duplicates?

Reply:
Only possible issue is that modifier order is no longer guaranteed. Are we okay with either making the user handle it, or adding helper fns to the types to do things like "get modifiers in a defined order"? Given that on the other side, even if the order isn't officially defined, if we make it ordered consumers will almost definitely rely on it.

Reply:
I think making it unordered is actually a *good* thing, I don't think people should rely on the order.